core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -952,7 +952,7 @@ class DynamicListenerConfig(server: KafkaBroker) extends BrokerReconfigurable wi
     oldAdvertisedListeners: Map[ListenerName, EndPoint],
     newAdvertisedListeners: Map[ListenerName, EndPoint]
   ): Boolean = {
-    if (oldAdvertisedListeners.size != newAdvertisedListeners.size) return true
+    if (oldAdvertisedListeners.sizeCompare(newAdvertisedListeners) != 0) return true
```

Comment:
FYI reviewers -
Comparison of two iterables. Reduces comparison time complexity from O(this.size + that.size) to O(this.size min that.size)